presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/TestReader.java

```diff
@@ -0,0 +1,209 @@
+        ImmutableList.Builder<SqlTimestamp> timestamps = new ImmutableList.Builder<>();
+        for (long value : epochMillisValues) {
+            timestamps.add(SqlTimestamp.fromMillis(3, value));
+        }
+
+        List<ObjectInspector> objectInspectors = singletonList(javaLongObjectInspector);
+        List<String> columnNames = ImmutableList.of("test");
+
+        ConnectorSession session = getHiveSession(new HiveConfig()
+                .setHiveStorageFormat(HiveStorageFormat.PARQUET)
+                .setUseParquetColumnNames(true));
+
+        try (ParquetTester.TempFile tempFile = new ParquetTester.TempFile("test", "parquet")) {
+            JobConf jobConf = new JobConf();
+            jobConf.setEnum(COMPRESSION, CompressionCodecName.SNAPPY);
+            jobConf.setBoolean(ENABLE_DICTIONARY, true);
+            jobConf.setEnum(WRITER_VERSION, PARQUET_1_0);
+
+            ParquetTester.writeParquetColumn(
+                    jobConf,
+                    tempFile.getFile(),
+                    CompressionCodecName.SNAPPY,
+                    ParquetTester.createTableProperties(columnNames, objectInspectors),
+                    getStandardStructObjectInspector(columnNames, objectInspectors),
+                    new Iterator<?>[] {epochMillisValues.iterator()},
+                    Optional.of(parquetSchema),
+                    false);
+
+            Iterator<SqlTimestamp> expectedValues = timestamps.build().iterator();
+            try (ConnectorPageSource pageSource = FileFormat.PRESTO_PARQUET.createFileFormatReader(session, HDFS_ENVIRONMENT, tempFile.getFile(), columnNames, ImmutableList.of(TIMESTAMP_MILLIS))) {
+                // skip a page to exercise the decoder's skip() logic
+                Page firstPage = pageSource.getNextPage();
+                for (int i = 0; i < firstPage.getPositionCount(); i++) {
+                    expectedValues.next();
+                }
+
+                while (!pageSource.isFinished()) {
+                    Page page = pageSource.getNextPage();
+                    if (page == null) {
+                        continue;
+                    }
+                    Block block = page.getBlock(0);
+
+                    for (int i = 0; i < block.getPositionCount(); i++) {
+                        assertEquals(TIMESTAMP_MILLIS.getObjectValue(session, block, i), expectedValues.next());
+                    }
+                }
+            }
+
+            assertFalse(expectedValues.hasNext());
+        }
+    }
+
+    @Test(dataProvider = "testTimestampMicrosDataProvider")
+    public void testTimestampMicros(HiveTimestampPrecision timestampPrecision, LocalDateTime expected)
+            throws Exception
+    {
+        ConnectorSession session = getHiveSession(new HiveConfig()
+                .setTimestampPrecision(timestampPrecision));
+
+        File parquetFile = new File(Resources.getResource("issue-5483.parquet").toURI());
```

Comment:
It would be nice to have a concise comment on steps needed to create a file (if we wanted to add some rows to it later on).

Reply:
i named the file after https://github.com/prestosql/presto/issues/5483 issue and the issue description begins with how to steps. is there a better way to not only write those steps down, but also make them findable by someone else?